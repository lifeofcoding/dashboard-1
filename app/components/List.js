// @flow
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import styles from './List.css';
import routes from '../constants/routes';

type Props = {
  add: (value) => void,
  incrementIfOdd: () => void,
  incrementAsync: () => void,
  remove: () => void,
  items: Array[]
  // textInput: string
};

export default class List extends Component<Props> {
  props: Props;
  
  constructor(props) {
    super(props);
    this.state = {
      textInput: ''
    };
  }
 
  onTextInputChange = (e) => {
    this.setState({textInput: e.target.value})
  }
  
  render() {
    const {
      add,
      incrementIfOdd,
      incrementAsync,
      remove,
      items,
      // textInput
    } = this.props;
    const { textInput } = this.state;
    const itemList = items.map((i, index) => {
      // console.log("i", i, "index", index)
      return (
      <div key={index}>
        <li>{i}</li>
        <button
            id="delete"
            className={styles.btn}
            onClick={() => {remove(index)}}
            data-tclass="btn"
            type="button"
          >
            <i className="fa fa-minus" />
          </button>
      </div> 
      )
    });
    return (
      <div>
        <div className={styles.backButton} data-tid="backButton">
          <Link to={routes.HOME}>
            <i className="fa fa-arrow-left fa-3x" />
          </Link>
        </div>
        <h2 className={styles.title}>
          TODO List:
        </h2>
        <div>
        <input 
        type="text" 
        onChange={this.onTextInputChange}
        onKeyUp={(e)=> { if(e.keyCode === 13) {add(textInput)} }}/>
        <span>You typed: <code>{textInput}</code> </span>
        </div>
        <div className={styles.btnGroup}>
        <button
            id="add"
            className={styles.btn}
            onClick={()=> { add(textInput)}}
            data-tclass="btn"
            type="button"
          >
            <i className="fa fa-plus" />
          </button>
          </div>
        <div className={`group-items ${styles.list}`} >
            {itemList}
        </div>
        {/* <div className={`counter ${styles.counter}`} data-tid="counter">
          {items}
        </div> */}
        {/* <div className={styles.btnGroup}>

        </div> */}
      </div>
    );
  }
}
