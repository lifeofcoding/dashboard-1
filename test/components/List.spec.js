import { spy } from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
// import Counter from '../../app/components/Counter';
import List from '../../app/components/List';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const actions = {
    add: spy(),
    // incrementIfOdd: spy(),
    addAsync: spy(),
    remove: spy()
  };
  const listofString = [1,2,3,4];
  const inputText = "Hello World";
  const component = shallow(<List items={listofString} inputTest={inputText} {...actions} />);
  return {
    component,
    actions,
    add: component.find('#add'),
    remove: component.find('#delete'),
    input: component.find('input'),
    p: component.find('.group-items'),
    l: listofString,
  };
}

describe('List component', () => {
  it('should should display list', () => {
    const { p, l } = setup();
    // expect(p.text()).toMatch(/^1$/);
    console.log("p text", p.text());
    expect(p.text()).toEqual(l.join(""));
  });

  it('should first button should call to add to list', ()=> {
    const { add, actions } = setup();
    add.at(0).simulate('click');
    expect(actions.add.called).toBe(true);
  })

  // it('should first button should call increment', () => {
  //   const { buttons, actions } = setup();
  //   buttons.at(0).simulate('click');
  //   expect(actions.increment.called).toBe(true);
  // });

  it('should match exact snapshot', () => {
    const { actions, l } = setup();
    const lst = (
      <div>
        <Router>
          <List counter={1} items={l} {...actions} />
        </Router>
      </div>
    );
    const tree = renderer.create(lst).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should remove button should call remove', () => {
    const { remove, actions } = setup();
    remove.at(0).simulate('click');
    expect(actions.remove.called).toBe(true);
  });

  it('type add new to do item after clickng add', () => {
    const { add, actions } =setup();
    add.at(0).simulate('click');
    expect(actions.add.called).toBe(true)
  })

  // it('should third button should call incrementIfOdd', () => {
  //   const { delete, actions } = setup();
  //   delete.at(0).simulate('click');
  //   expect(actions.incrementIfOdd.called).toBe(true);
  // });

  // it('should fourth button should call incrementAsync', () => {
  //   const { buttons, actions } = setup();
  //   buttons.at(3).simulate('click');
  //   expect(actions.incrementAsync.called).toBe(true);
  // });
});
