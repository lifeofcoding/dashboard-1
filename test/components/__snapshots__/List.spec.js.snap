// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should match exact snapshot 1`] = `
<div>
  <div>
    <div
      className="backButton"
      data-tid="backButton"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <i
          className="fa fa-arrow-left fa-3x"
        />
      </a>
    </div>
    <h2
      className="title"
    >
      TODO List:
    </h2>
    <div>
      <input
        onChange={[Function]}
        type="text"
      />
      <span>
        You typed: 
        <code>
          
        </code>
         
      </span>
    </div>
    <div
      className="group-items list"
    >
      <div>
        <li>
          1
        </li>
        <button
          className="btn"
          data-tclass="btn"
          id="delete"
          onClick={[Function]}
          type="button"
        >
          <i
            className="fa fa-minus"
          />
        </button>
      </div>
      <div>
        <li>
          2
        </li>
        <button
          className="btn"
          data-tclass="btn"
          id="delete"
          onClick={[Function]}
          type="button"
        >
          <i
            className="fa fa-minus"
          />
        </button>
      </div>
      <div>
        <li>
          3
        </li>
        <button
          className="btn"
          data-tclass="btn"
          id="delete"
          onClick={[Function]}
          type="button"
        >
          <i
            className="fa fa-minus"
          />
        </button>
      </div>
      <div>
        <li>
          4
        </li>
        <button
          className="btn"
          data-tclass="btn"
          id="delete"
          onClick={[Function]}
          type="button"
        >
          <i
            className="fa fa-minus"
          />
        </button>
      </div>
    </div>
    <div
      className="btnGroup"
    >
      <button
        className="btn"
        data-tclass="btn"
        id="add"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa fa-plus"
        />
      </button>
      <button
        className="btn"
        data-tclass="btn"
        id="delete"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa fa-minus"
        />
      </button>
    </div>
  </div>
</div>
`;
